src/hotspot/share/opto/doCall.cpp

```diff
@@ -111,6 +111,14 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
     log->end_elem();
   }
 
+  // Try blackholing a method
+  if (callee->is_loaded() &&
```

Comment:
The check can be turned into a property on `ciMethod` and the result cached there.

Reply:
Yes, but only partially, because C1 and C2 check their own directive sets, and C2 also checks for matching rule. But it does skip unloaded methods more cleanly if we move the first two checks to `ciMethod`. Fixed.